perun/view/flamegraph/flamegraph.py

```diff
@@ -69,35 +66,31 @@ def draw_flame_graph_difference(
 
 
 def draw_flame_graph(
-    profile: Profile,
-    width: int = 1200,
-    max_trace: int = 0,
-    max_resource: float = 0.0,
-    title: str = "",
-    profile_key: str = "amount",
-    minimize: bool = False,
+    flame_data: list[str],
+    title: str,
+    units: str = "samples",
+    img_width: int = 1200,
+    fg_min_width: str = "1",
+    fg_max_trace: int = 0,
+    fg_max_resource: float = 0.0,
 ) -> str:
-    """Draw Flame graph from profile.
+    """Draw Flame graph from flame data.
 
         To create Flame graphs we use perl script created by Brendan Gregg.
         https://github.com/brendangregg/FlameGraph/blob/master/flamegraph.pl
 
-    :param profile: the memory profile
-    :param width: width of the graph
-    :param profile_key: key for which we are constructing profile
-    :param title: if set to empty, then title will be generated
+    :param flame_data: the data to generate the flame graph from
+    :param title: title of the flame graph
+    :param units: the units of the flame graph data
+    :param img_width: width of the graph
+    :param fg_min_width: minimum width of the flame graph rectangles that will be drawn
+    :param fg_max_trace: maximum length of traces that are being drawn
+    :param fg_max_resource: maximum number of samples collected
     """
     # converting profile format to format suitable to Flame graph visualization
-    flame = convert.to_flame_graph_format(profile, profile_key=profile_key, minimize=minimize)
-
-    header = profile["header"]
-    profile_type = header["type"]
-    cmd, workload = (header["cmd"], header["workload"])
-    title = title if title != "" else f"{profile_type} consumption of {cmd} {workload}"
-    units = mapping.get_unit(mapping.get_readable_key(profile_key))
-
+    # flame = convert.to_flame_graph_format(profile, profile_key=profile_key, minimize=minimize)
```

Comment:
Why is this commented out here?

Reply:
Good catch, this comment shouldn't be there. I'll remove it.